### PR TITLE
Samples documentation fixes and enhanchements

### LINKED ISSUE #801: List of open issues with Antora documentation
With PRs #799 and #800 the Antora documentation for the samples is pretty much ready for release.

This is a meta issue to track remaining issues with the documentation part of the samples.

I noticed a few issues though that we need/should address at some point:

- Source file links in the framework documentation don't work
- Images for the hpp samples aren't displayed, probably because they have relative links that afaik won't work with Antora (Texture mipmap generation, pipeline cache, swapchain images)
- ~~Formatting for some of the hpp sample readmes is off because of wrong indentation levels (=== before =)~~ Fixed with PR #808
- Screenshot for the dynamic line rasterization sample is not displayed
- Some of the internal links between pages in the project aren't working (see output when building the site)
- We have a lot of links to the spec registry, which should be changed to link to the docs site spec project instead
- ~~Some files have explicit table of contents (e.g. the build guide), which won't work with antora and is also not necessary, as antora always shows a table of contents at the right hand side of a page~~ Fixed with PR #808
- Some inter-documentation links are broken, e.g. in the readme for the "Appropriate use of render pass attachments" sample

### LINKED ISSUE #824: Antora build is finding a bunch of broken links
This might duplicate #820, not sure - see e.g. https://github.com/KhronosGroup/Vulkan-Site/actions/runs/6452777410/job/17525321493#step:20:164 :

> [12:07:15.309] ERROR (asciidoctor): target of xref not found: docs/pull_request_template.md  
>    file: /home/runner/work/Vulkan-Site/Vulkan-Site/Vulkan-Samples/antora/modules/ROOT/pages/CONTRIBUTING.adoc  
>    source: /home/runner/work/Vulkan-Site/Vulkan-Site/Vulkan-Samples (branch: main <worktree> | start path: antora)

FWIW I have found it *immensely* helpful with the API spec to make all include:: / image: paths absolute, parameterized by asciidoctor attributes with the shared path prefix (e.g. include::{chapters}/filename.adoc[]). Then for Antora I make all of those attributes point into partials/ or images/ and symlink the actual top-level files from partials/ into pages/.

### LINKED ISSUE #898: Missing link in extensions readme
https://docs.vulkan.org/samples/latest/samples/extensions/README.html contains a missing link to https://docs.vulkan.org/samples/latest/samples/README.adoc#extension-samples

As the samples are now navigatable via docs.vulkan.org listing them might still be useful for people who search directly in github.

### LINKED ISSUE #903: Links to ray tracing samples broken
While fixing #898 and reworking how samples listings are generated for the documentation site, I noticed that the links to some of the ray tracing samples are broken. Probably because they have been renamed some time ago.

I'll try to fix this.

### PR DESCRIPTION
**Pull request by SaschaWillems**
## Description

I went through all of the outstanding issues with regards to the samples documentation generated for https://docs.vulkan.org and hopefully fixed all open problems with this PR. Aside from fixing things, this PR also improves documentation in a few spots. Once merged documentation for our samples should be perfect now.

Bugfixes:
- Fixed ~30 broken links, both to documents and images (antora build is now 100% clean for me)
- Added in some referenced documents that were simply not present

Enhanchements:
- Split the samples list into one file per folder (api, performance, extension, etc.), so that every sub folder in the docs site now shows the list of samples for this exact folder
- The samples overview listing is now created by include the above files, so it looks just as a before
- Added links to a few documents in the navigation that were there but not accessible through it yet

Fixes #903 
Fixes #898 
Fixes #824 
Fixes #801

**Note: This is a pure documentation fix**

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Copyright-Notice-and-License-Template)
- [n/a] I have commented any added functions (in line with Doxygen)
- [n/a] I have commented any code that could be hard to understand
- [n/a] My changes do not add any new compiler warnings
- [n/a] My changes do not add any new validation layer errors or warnings
- [n/a] I have used existing framework/helper functions where possible
- [n/a] My changes do not add any regressions
- [n/a] I have tested every sample to ensure everything runs correctly
- [n/a] This PR describes the scope and expected impact of the changes I am making

### CODE REVIEW

**Review by tomadamatkinson — COMMENTED** [review]
Mega! Thanks @SaschaWillems for putting in the graft on this one

Is there a quick script we could right to make sure all the links work? Looks like all the paths are relative so we could make a small CI task to make life easier in the future?

Happy to help here

**Comment by SaschaWillems** [comment]
Running antora to build the docs site will output errors. If you pipe that output you get JSON that can easily be parsed. Though I'm not sure if we can do this in this repo alone, as the samples are only a part of the whole docs repo. So if at all, we should do it in the doc site repo. Though I think Jon does that anyway.

Adding it to this repo for the CI would probably make life hard for contributors, as you'd need to know how to build an antpra project and how to fix things.

**Comment by tomadamatkinson** [comment]
Makes sense, it's low priority if we did pursue it. Jsut thought we might be able to whip something up in 30 minutes which might help save yourself a bit of effort in the long run

Happy to stay as is for now 😃

**Review by gary-sweet — DISMISSED** [review]

File: samples/README.adoc [hunk 1]
```diff
@@ -16,6 +16,12 @@
 - limitations under the License.
 -
 ////
+ifndef::performance_samplespath[:performance_samplespath: ./performance/]
+ifndef::extenstion_samplespath[:extenstion_samplespath: ./extensions/]
```
Comment (asuessenbach) [inline samples/README.adoc]: Is that a typo: `extenstion_samplespath`?

Comment (SaschaWillems) [reply samples/README.adoc]: Indeed. Thanks for noticing!

**Review by gary-sweet — DISMISSED** [review]

**Comment by SaschaWillems** [comment]
@marty-johnson59 : Would be great of we could merge this on the next samples/doc call before we merge any other new sample PRs to avoid merge conflicts.

**Review by asuessenbach — APPROVED** [review]

**Pull request merged**